webapp_v2/src/components/settings/Tokens/dialog/TokenFormDialog/TokenForm.tsx

```diff
@@ -0,0 +1,227 @@
+  // JSX
+  ////////////////////////////////////////////////////////////////
+
+  return (
+    <>
+      {/* Name */}
+      {!onlyPermissions && (
+        <>
+          <TextField
+            sx={{ mx: 0 }}
+            label={t("main:name")}
+            error={!!errors.name}
+            helperText={errors.name?.message}
+            required
+            fullWidth
+            {...register("name", {
+              required: t("main:form.field.required") as string,
+            })}
+          />
+          <FormControlLabel
+            control={
+              <Checkbox defaultChecked={false} {...register("linkToUser")} />
+            }
+            label={t("settings:linkTokenLabel")}
+          />
+        </>
+      )}
+      {/* Permissions */}
+      <Paper
+        sx={{
+          p: 2,
+          mt: 2,
+          backgroundImage:
+            "linear-gradient(rgba(255, 255, 255, 0.05), rgba(255, 255, 255, 0.05))",
+        }}
+      >
+        <Typography>{t("settings:permissionsLabel")}</Typography>
+        {isGroupsLoading && (
+          <Box
+            sx={{
+              display: "flex",
+              justifyContent: "center",
+              alignContent: "center",
+              mt: 2,
+            }}
+          >
+            <CircularProgress color="inherit" />
+          </Box>
+        )}
+        {filteredAndSortedGroups.length > 0 && (
+          <>
+            <Box sx={{ display: "flex", alignItems: "center", mt: 2 }}>
+              <FormControl sx={{ mr: 2, flex: 1 }} size="small">
+                <InputLabel id={groupLabelId}>{t("settings:group")}</InputLabel>
+                <Select
+                  labelId={groupLabelId}
+                  label={t("settings:group")}
+                  defaultValue=""
+                  onChange={handleGroupChange}
+                >
+                  {filteredAndSortedGroups.map((group) => (
+                    <MenuItem key={group.id} value={group.id}>
+                      {group.name}
+                    </MenuItem>
+                  ))}
+                </Select>
+              </FormControl>
+              <Button
+                variant="contained"
+                size="small"
+                disabled={!allowToAddPermission}
+                onClick={() => {
+                  append({ group: selectedGroup, type: RoleType.READER });
+                }}
+              >
+                {t("settings:addButton")}
+              </Button>
+            </Box>
+            <List>
+              {fields.map((field, index) => (
+                <ListItem
+                  key={field.id}
+                  secondaryAction={
+                    <>
+                      <Controller
+                        control={control}
+                        name={`permissions.${index}.type`}
+                        render={({ field }) => (
+                          <Select variant="standard" {...field}>
+                            {ROLE_TYPE_KEYS.map((key) => (
```

Comment:
this should be filtered with permissions below or equal (ROLE_TYPE_KEYS are integers) to the current user's permission to that group (not needed when admin).